wgsl/index.bs

```diff
@@ -5481,7 +5597,7 @@ allows them to naturally use values defined in the loop body.
 <div class='syntax' noexport='true'>
   <dfn for=syntax>for_header</dfn> :
 
-    | ( [=syntax/variable_statement=] | [=syntax/assignment_statement=] | [=syntax/func_call_statement=] ) ? [=syntax/semicolon=] [=syntax/expression=] ? [=syntax/semicolon=] ( [=syntax/assignment_statement=] | [=syntax/func_call_statement=] ) ?
+    | ( [=syntax/variable_statement=] | [=syntax/assignment_statement=] | [=syntax/func_call_statement=] ) ? [=syntax/semicolon=] [=syntax/expression=] ? [=syntax/semicolon=] ( [=syntax/assignment_statement=] | [=syntax/complex_assignment_statement=] | [=syntax/func_call_statement=] ) ?
```

Comment:
What's the reason for not allowing a complex assignment as a for loop initializer?

Having now implemented this in Tint, it seems a little awkward that compound assignment is a different grammar rule to regular assignment. We can't actually parse them completely independently. If you start parsing `assignment_statement` and then hit something that isn't `'='`, you don't want to backtrack all the way back past the LHS expression to try again for `complex_assignment_statement`, so we instead parse the LHS and then check if we have a `'='` or a `'{+=, -=, ...}'`. I feel like it'd be cleaner to use the same grammar rule for both (and allow both here in the for loop initializer), which would match how we parse it.